### PR TITLE
Rename DatasourceId path parameter

### PR DESCRIPTION
**Pull request by papagian**
We hope that this would be less confusing for grafana users. 

Its value should be `grafana` for requests to be handled by grafana and the numeric datasource id for requests to be forwarded to a lotex datasource.

The chosen name `Recipient` is discussable.

**Update**
I have added a 0ed8a62ac063dbc0ad23c56d251d6e2234884ef8 which is also [required](https://github.com/grafana/grafana/pull/32208#discussion_r598828511) for grafana/grafana#32208.

### CODE REVIEW

**Review by domasx2 — APPROVED** [review]
`Recipient` makes me think of emails :D 

Still gonna be int, === datasource.id, right? or will it take datasource.name as well?

**Review by owen-d — APPROVED** [review]
Recipient sounds good, but we'll need to ensure a datasource name won't conflict with this reserved "grafana"key.

**Comment by papagian** [comment]
> `Recipient` makes me think of emails :D
> 
> Still gonna be int, === datasource.id, right? or will it take datasource.name as well?

In the suggested [implementation](https://github.com/grafana/grafana/pull/32208) if it equals the reserved "grafana" key is handled by the grafana otherwise is expected to be convertible to integer and that this number corresponds to a lotex datasource id.
Hence, identifying the datasource by the name is not supported. It's feasible though but it would complicate things a bit
as @owen-d mentioned if the datasource name conflicts with the reserved key. Therefore, I would suggest keeping it as is unless it's absolutely necessary.

**Comment by domasx2** [comment]
> > `Recipient` makes me think of emails :D
> > Still gonna be int, === datasource.id, right? or will it take datasource.name as well?
> 
> In the suggested [implementation](https://github.com/grafana/grafana/pull/32208) if it equals the reserved "grafana" key is handled by the grafana otherwise is expected to be convertible to integer and that this number corresponds to a lotex datasource id.
> Hence, identifying the datasource by the name is not supported. It's feasible though to support it but it would complicate things a bit
> as @owen-d mentioned if the datasource name conflicts with the reserved key. Therefore, I would suggest keeping it as is unless it's absolutely necessary.

That's cool, just wanted to be sure. Thanks!

File: pkg/api/api.go [hunk 1]
```diff
@@ -31,7 +31,21 @@ type Ack struct{}
 type Backend int
 
 const (
-	GrafanaBackend Backend = iota
+	GrafanaBackend Backend = iota + 1
```
Comment (owen-d) [inline pkg/api/api.go:34]: Why did you change this to be `iota + 1`?
edit: to not conflict with the zero value -- that's fine.

**Review by owen-d — APPROVED** [review]
